server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -875,4 +511,453 @@ public void forcingDelayedMailsDeliveryShouldActuallyChangePropertyOnMails() thr
                 assertThat(delivery.get()).isBefore(ZonedDateTime.now());
             });
     }
+
+    @Nested
+    class DeleteMail {
```

Comment:
Can we have Nested classes for the other REST operations as well?

Reply:
This Nested class is in my opinion still too long.

If a Nested class can have Nested class, then we should introduce the following sub-categories:

 - **DataValidation** (all 400 errors and asserts on return codes)
 - **SideEffects** (every time we assert the content of the queue)
 - **HttpBodies** (every time we assert the content of the cody)

If some other categroies makes sens then we should add them. I do believe it should help us to structure our tests... WDYT ?

Reply:
Yeah, I think so. And the sub-categories in this `DeleteMail` class, I prefer some easy to understand circumstance nested class name:
- AssertFail
- AssertComplete
- AssertBody
- AssertQueueAfterDelete

Reply:
oh but your suggestion is more applicable in other test 👍

Reply:
I don't understand what **AssertFail** refer to. I would propose **AssertValidationFails**.

**AssertHttpBody** will be more helpful.